lis/interp/map_utils.F90

```diff
@@ -681,7 +681,8 @@ SUBROUTINE llij_gauss(lat, lon, proj, i, j)
       slon360 = proj%lon1
     ENDIF
     deltalon = lon360 - slon360
-    IF (deltalon .LT. 0) deltalon = deltalon + 360.
+    ! MB: only + 360 when rounding of deltalon/proj%dlon + 1. does not give 1
+    IF (deltalon .LT. (-0.5*proj%dlon)) deltalon = deltalon + 360.
```

Comment:
Not sure about this fix: e.g. if deltalon > -0.5*proj%dlon, but still negative, then deltalon remains negative and "i" wil be less than 1 and I don't know if i<1 can ever be possible. "i" is a real number and can have decimal values, but it is perhaps not guaranteed that "i" is rounded off everywhere in the code to 1.

Reply:
I understand that my very brief PR description was not sufficient to convince. Sorry. 

With the 30arcsec grid of the harmonized soil map (dx,dy, 0.0083333) we encountered that we only have 'stripe artifacts' in the increment output, but not in the obs and innov output. And this only in parallel and not in single core mode. 

Tracing this back in the code, I found that it happens in the subroutine enkf_increments of the enkf_Mod.F90

          call LIS_surfaceModel_DAmapTileSpaceToObsSpace(&
               n, k, &
               tileid, st_id, en_id)

          if(st_id.lt.0.or.en_id.lt.0) then
             assim = .false.

Because st_id and en_id are <0 we don't assimilate also there are innovs. i.e. something goes wrong in  LIS_surfaceModel_DAmapTileSpaceToObsSpace. The latter calls LIS_lsm_DAmapTileSpaceToObsSpace which calls llij_latlon through latlon_to_ij.

llij_latlon handles the subdomain to domain mapping (indexing) for the subdomains of each parallel process. llij_latlon returns domain indices in *real* format and
whenever latlon_to_ij is called in LIS, i and j are converted to integer with nint (rounding) to be used as indices.
For the harmonized grid, deltalon = lon360 – slon360 in latlon_to_ij yielded for some longitudes a difference of -0.0001. Because this is slightly negative, it eventually yielded the max longitude index because of
IF (deltalon .LT. 0) deltalon = deltalon + 360.
Because this is outside of the global domain, and IF statements sets i and j to -1 in LIS_lsm_DAmapTileSpaceToObsSpace which causes assim=.false. in  the subroutine enkf_increments (see above). 
The 0W issue in latlon_to_ij can be addressed by only adding 360 when (deltalon .LT. (-0.5*proj%dlon)). This would give the smalles index (i.e. 1) when deltalon is less than half of a grid cell).  

@s-scherrer : Since you worked on the +360, could you perhaps have a look on it as well. Thx.

PS: the term 'dateline' I did adopt from the existing lis code. I agree that calling it 0W issue makes more sense.

Reply:
OK as a patch for now, but I think the problem is rather in slight round-off issues at the input side.